storage.go

```diff
@@ -4,37 +4,39 @@ import (
 	"context"
 )
 
-// Storage represents the abstraction for underlying storage backends. A storage
-// implementation does not care about specific data types. All the
-// storage cares about are key-value pairs. Code making use of the storage
-// have to take care about specific types they care about them self.
+// KV is a key-value pair.
+type KV struct {
+	// Key should take a format of path separated by slashes "/". E.g.
+	// "/a/b/c". Storage implementations should validate the key before
+	// storing them. Key sanitized before inserting to the storage. I.e.
+	// leading slash can be added and trailing slash can be removed. E.g.
+	// "/a/b/c", "a/b/c/", "a/b/c/", and "/a/b/c/" represent the same key.
+	Key string
+	// Val is an arbitrary value stored under the Key.
+	Val string
+}
+
+func NewKV(key, val string) KV {
+	return KV{
+		Key: key,
+		Val: val,
+	}
+}
+
+// Storage represents the abstraction for underlying storage backends.
 type Storage interface {
-	// Create is deprecated in favour of Put. Its semantics is unspecified
-	// when the value of the key does not exist.
-	//
-	// Create stores the given value under the given key. Keys and values
-	// might have specific schemes depending on the specific storage
-	// implementation.  E.g. an etcd storage implementation will allow keys
-	// to be defined as paths: path/to/key. Values might be JSON strings in
-	// case the storage implementation wants to store its data as JSON
-	// strings.
-	Create(ctx context.Context, key, value string) error
 	// Put stores the given value under the given key. If the value
-	// under the key already exists Put overrides it. Keys and values might
-	// have specific schemes depending on the specific storage
-	// implementation.  E.g. an etcd storage implementation will allow keys
-	// to be defined as paths: path/to/key. Values might be JSON strings in
-	// case the storage implementation wants to store its data as JSON
-	// strings.
-	Put(ctx context.Context, key, value string) error
+	// under the key already exists Put overrides it.
+	Put(ctx context.Context, kvs ...KV) error
```

Comment:
I removed some comments, which aren't true anymore. I made effort to make implementations consistent, and hope this is desired . That way we can swap implementation without changing logic. So we want keys to be in specific format. This is described in `KV.Key`.